File: scripts/time.py
#!bin/python3

# usage: python3 time.py <number of iterations> <program> <arguments>

# get the variables from the command line
import sys
iterations = int(sys.argv[1])
program = sys.argv[2]
arguments = sys.argv[3:]


# run a program for a number of iterations with the arguments and print the time taken
import time
import subprocess
start = time.time()
for i in range(iterations):
	subprocess.call([program] + arguments)
end = time.time()
print(end - start)

File: scripts/time.lua
-- usage: krait -e time.lua <number of iterations> <program> <arguments>
-- note: this is pseudocode (for possible future implementation)

-- get the command line arguments
local args = krait.args
local iterations = tonumber(args[1])
local program = args[2]
local program_args = {}

-- get the program arguments
for i = 3, #args do
	program_args[i - 2] = args[i]
end

-- run the program
local start = krait.sys.time()
for i = 1, iterations do
	krait.program.exec(program, program_args)
end
local finish = krait.sys.time()

-- print the time
print("time: " .. (finish - start) .. " seconds")